packages/components/src/components/checkbox-wrapper/checkbox-wrapper-styles.ts

```diff
@@ -32,13 +44,39 @@ export const getComponentCss = (
               `<path fill="${indeterminateIconColor}" d="m20,11v2H4v-2h16Z"/>`
             ),
           },
-          ...(!isDisabled && {
-            '&(input:focus)::before': {
-              borderRadius: borderRadiusMedium,
-            },
-          }),
+          ...(!isDisabled &&
+            !isLoading && {
+              '&(input:focus)::before': {
+                borderRadius: borderRadiusMedium,
+              },
+            }),
         }),
       },
+
+      spinner: {
+        width: fontLineHeight,
+        height: fontLineHeight,
+        pointerEvents: 'none',
+        position: 'absolute',
+        fontFamily, // needed for correct width and height definition and for correct positioning
+        fontSize: '1rem', // needed for correct width and height definition and for correct positioning
+        top: '50%',
+        left: `calc(${fontLineHeight}/2 + 2px)`,
+        transform: 'translate(-50%, -50%)',
+        border: `2px solid ${disabledColor}`,
+        borderRadius: borderRadiusSmall,
+        ...(isChecked && {
+          background: disabledColor,
+        }),
+        ...(isLoading
```

Comment:
`opacity: isLoading ? 1 : 0` would be easier to read and uses less bytes.

Reply:
fixed it